### DISCUSSION TITLE
Certain utility classes not working and some for only certain numbers

### DISCUSSION THREAD

**Discussion by vonkoff** [tailwindlabs/tailwindcss#9622] [Help] [answered]
So, I'm trying to truncate my lines on this website for when people are looking in mobile. Sometimes the length of the job name goes into three lines which overflows into the job card and it doesn't look too pretty.
I've tried using the Tailwind plugin for line-clamp. No good. Also, tried webkit LineClamp inline with React. No good.
Here is the code on my end that I'm using at [goodunionjobs.com](https://goodunionjobs.com/)  . you can also check most of the code through there. 
```ts
            <div className="flex-auto font-Atkinson">
              <p className="text-sm font-normal leading-tight text-ellipsis ">
                <Highlight attribute="employer" hit={hit} />
              </p>
              <p className="sm:text-sm text-lg leading-tight sm:leading-4 text-ellipsis">
                <Highlight
                  attribute="job"
                  hit={hit}
                  style={{ WebkitLineClamp: 3 }}
                />
              </p>
              {hit.salary === '' ? (
                <span className="hidden"></span>
              ) : (
                <span
                  className="sm:text-xs text-sm font-bold whitespace-nowrap px-1 pt-0.5 rounded-sm 
                  bg-gray-200 border-solid border-2 border-black text-black"
                >
                  💰{hit.salary}
                </span>
              )}
              {hit.state === '' ? (
                <span className="hidden"></span>
              ) : (
                <span
                  className="sm:text-xs text-sm font-bold whitespace-nowrap px-1 pt-0.5 rounded-sm 
                  bg-gray-200 mx-1 border-solid border-2 border-black text-black"
                >
                  📌{hit.state}
                </span>
              )}
              {hit.employmentType === '' ? (
                <span className="hidden"></span>
              ) : (
                <span
                  className="sm:text-xs text-sm font-bold whitespace-nowrap px-1 pt-0.5 rounded-sm 
                  bg-gray-200 border-solid border-2 border-black text-black"
                >
                  🕑{hit.employmentType}
                </span>
              )}
            </div>
```
It doesn't even work when I try changing the development to only use the <p> tag like how I did below (I also attached a picture).
![image](https://user-images.githubusercontent.com/16548992/197020911-adc9c562-6046-4adc-a885-cad0967573e6.png)
```ts
              <p className="text-sm font-normal leading-tight text-ellipsis ">
                <Highlight attribute="employer" hit={hit} />
              </p>
              <p
                className="sm:text-sm text-lg leading-tight sm:leading-4 text-ellipsis line-clamp-1"
                style={{ WebkitLineClamp: 2 }}
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
                eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut
                enim ad minim veniam, quis nostrud exercitation ullamco laboris
                nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor
                in reprehenderit in voluptate velit esse cillum dolore eu fugiat
                nulla pariatur. Excepteur sint occaecat cupidatat non proident,
                sunt in culpa qui officia deserunt mollit anim id est laborum.
                {/* <Highlight attribute="job" hit={hit} /> */}
              </p>
```

### Another weird issue regarding some utility classes
I can only use margins or padding like `m-4` or `p-4`. If I try a different number like 2,3, etc then it doesn't work. It works for some numbers, but mostly not a lot of them. I've kind of noticed this happens in certain parts of my code and I am not exactly sure when the issue started because I have been kind of avoiding it. 
Also, seeing now that text-align doesn't work when using it on a `<p>` tag

**They should work because if I add** `style={{ textAlign: 'center' }}` or ` <main style={{ margin: '25px 25px 25px 25px' }}>` to my react code then it works!

**Comment by wongjn** [comment, marked as answer]
I applied the CSS rules that would be applied by `line-clamp-` to the `<p>` tag in devtools on the live site and it seems to work – perhaps verify that the class is being applied?

With regards to some classes not working, ensure that the files that you use the class in is covered by the `content` file globs in the Tailwind configuration. Also check that the spacing values exist in the *resolved* Tailwind configuration.

**Reply by vonkoff** [reply]
Ah ok! I see the package.json I was running something with `&&` even though I was using `concurrently`. I think that was the issue and getting rid off `&&`  and the code before it fixed it. Don't know why some things worked though and others did not when I was running things before.
